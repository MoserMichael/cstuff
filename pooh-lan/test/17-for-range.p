# co-routines / generators as used with for statement

sub myrange( from, to )
   i = from
   while i <= to

     # yield function puts this thread to sleep and will wake up the caling thread
     # it also returns a value to the calling thread.

     threadyield0( ~yieldval i )

     i = i + 1
   end
end

#
# Invoke the co-routine via the for loop syntax
#

for value myrange( ~from 1 ~to 10 )
  println( ~msg 'thread returned [ value ]' )
end

