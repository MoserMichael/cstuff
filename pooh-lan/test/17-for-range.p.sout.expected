thread returned 1
thread returned 2
thread returned 3
thread returned 4
thread returned 5
thread returned 6
thread returned 7
thread returned 8
thread returned 9
thread returned 10
